### PR TITLE
Factor out logging code from Buildozer class

### PR DESCRIPTION
**Pull request by Julian-O**
First step toward #1629.

Code related to logging moved into its own class/file.
References (including in targets and tests) updated.
Unit tests separated out, and enhanced to test more conditions.
Trivial indenting/grammar issues in osx.py fixed.
No-op exception handling code removed.

### CODE REVIEW

File: buildozer/targets/osx.py [hunk 1]
```diff
@@ -20,14 +22,14 @@ def ensure_sdk(self):
                 join(self.buildozer.platform_dir, 'kivy-sdk-packager-master')):
             self.buildozer.info(
```
Comment (Julian-O) [inline buildozer/targets/osx.py:23]: Missed this one.

**Review by misl6 — CHANGES_REQUESTED** [review]
Hi @Julian-O !

In `buildozer/scripts/remote.py` references have not been updated.
Can you please take care of it?

**Review by misl6 — CHANGES_REQUESTED** [review]
Seems we still have two old references:
https://github.com/kivy/buildozer/blob/0851a101b5e301803a42ed5f5562e6910bff116b/buildozer/scripts/client.py#L19
https://github.com/kivy/buildozer/blob/0851a101b5e301803a42ed5f5562e6910bff116b/buildozer/scripts/remote.py#L274

Unfortunately I missed them from the previous review.
(That part will likely need a huge cleanup, but makes sense to keep it updated ATM)

**Comment by Julian-O** [comment]
These sneaky ones created a Buildozer() object just to call logging! I missed them in my text searches. That left me red-faced.  I should have caught these, sorry.

**Review by misl6 — APPROVED** [review]
LGTM. Thank you!

(The CI pipeline is failing, but is a known issue with Cython 3.0.0)

**Pull request merged**